pkgs/applications/misc/hello/salve-mundi.nix

```diff
@@ -0,0 +1,98 @@
+/*
+  This file showcases various techniques and features of Nixpkgs, whereas the
+  regular hello serves as more of an introductory package definition.
+  Salve mundi is latin for hello world.
+*/
+{ callPackage
+, lib
+, stdenv
+, fetchurl
+, nixos
+, testers
+, salve-mundi
+, makeWrapper
+, runCommand
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "salve-mundi";
+  version = "2.12.1";
+
+  src = fetchurl {
+    url = "mirror://gnu/hello/hello-${finalAttrs.version}.tar.gz";
+    sha256 = "sha256-jZkUKv2SV28wsM18tCqNxoCZmLxdYH2Idh9RLibH2yA=";
+  };
+
+  doCheck = true;
+
+
+  /* Wrap the program */
+
+  postInstall = lib.optionalString (finalAttrs.passthru.greeting != null) ''
+    wrapProgram $out/bin/hello --append-flags --greeting=${lib.escapeShellArg (lib.escapeShellArg finalAttrs.passthru.greeting)}
+  '';
+  nativeBuildInputs = [ makeWrapper ];
+  passthru.greeting = "SALVE MUNDI";
+
+
+  /* Tidy up the package attributes and make them useful */
+
+  __cleanAttrs = true;
+  passthru.exe = lib.getExe finalAttrs.finalPackage;
+  # We allow `hello.src` to be used in tests and examples, despite __cleanAttrs
+  passthru.src = finalAttrs.src;
+
+
+  /* Strict deps enforce a separation of concerns that also benefits cross compilation
+     and, for shells, shell completions support via nativeBuildInputs */
+
+  strictDeps = true;
+
+
+  /* A fairly extensive suite of extra tests that we like to hold either for
+     the package in the Nixpkgs package set, or even for all possible overrides
+     of the package. */
+
+  passthru.tests = {
+    version = testers.testVersion { package = salve-mundi; };
+
+    # We use Nixpkgs attributes instead of `finalAttrs.finalPackage` here
+    # because overriding is not supported. Running the test on an overridden
+    # finalPackage wouldn't work, and is a bit unnecessary anyway.
+    invariant-under-noXlibs =
+      testers.testEqualDerivation
+        "hello must not be rebuilt when environment.noXlibs is set."
+        salve-mundi
+        (nixos { environment.noXlibs = true; }).pkgs.salve-mundi;
+
+    run = runCommand "salve-mundi-test-run" {
+      nativeBuildInputs = [ finalAttrs.finalPackage ];
+    } ''
+      diff -U3 --color=auto <(hello) <(echo 'SALVE MUNDI')
+      touch $out
+    '';
+
+    restore-default-greeting = callPackage ./test.nix {
+      hello = finalAttrs.finalPackage.overrideAttrs (o: {
+        passthru = o.passthru // {
+          greeting = null;
+        };
+      });
+    };
```

Comment:
What's this for?

Reply:
not everyone speaks latin :smile:

Reply:
That is a good point. It could come across as elitist, especially in this context now that I think about it more.
Something like Esperanto would be nicer, or Lojban, or Tolkien's Elvish. Or maybe it should just be `hello_with_bells_and_whistles`, although I don't know if `_` for variations should be followed by snake case or camelCase.
Why is this hard...

May the first confirmation win:
 - `suilad-ambar`: is this a correct Elvish package name?
 - `coi-munje`: is this a correct Lojban package name?

I don't like the Esperanto version because it's basically Romance language vocabulary. Nothing wrong with that, but if it's not going to be English, I don't want to bias it. Sue me. ;)